pennylane_lightning/core/src/simulators/lightning_qubit/gates/DynamicDispatcher.hpp

```diff
@@ -433,6 +489,32 @@ template <typename PrecisionT> class DynamicDispatcher {
                                            func);
     }
 
+    /**
+     * @brief Register a new sparse matrix operation. Can pass a custom
+     * kernel.
+     * @param mat_op Sparse matrix operation
+     * @param kernel Kernel
+     * @param func Function pointer for the sparse matrix operation
+     */
+    void registerSparseMatrixOperation(SparseMatrixOperation mat_op,
```

Comment:
Correct me if I'm wrong, we wouldn't need these methods if we reuse the existing kernel maps for multi-qubit ops?